apps/dashboard/src/components/number-input-with-select.tsx

```diff
@@ -0,0 +1,97 @@
+import { cn } from '@/utils/ui';
+import { FormControl, FormField, FormItem, FormMessagePure } from '@/components/primitives/form/form';
+import { Input } from '@/components/primitives/input';
+import { InputFieldPure } from '@/components/primitives/input';
+import { Select, SelectContent, SelectItem, SelectTrigger, SelectValue } from '@/components/primitives/select';
+import { useFormContext } from 'react-hook-form';
+import { useMemo } from 'react';
+
+type InputWithSelectProps = {
+  inputName: string;
+  selectName: string;
+  options: string[];
+  defaultOption?: string;
+  className?: string;
+  placeholder?: string;
+  isReadOnly?: boolean;
+};
+
+export const NumberInputWithSelect = (props: InputWithSelectProps) => {
+  const { className, inputName, selectName, options, defaultOption, placeholder, isReadOnly } = props;
+
+  const { getFieldState, setValue, getValues, control } = useFormContext();
+
+  const amount = getFieldState(`${inputName}`);
+  const unit = getFieldState(`${selectName}`);
+  const error = amount.error || unit.error;
+
+  const defaultSelectedValue = useMemo(() => {
+    return defaultOption ?? options[0];
+  }, [defaultOption, options]);
+
+  const handleChange = (value: { input: number; select: string }) => {
+    // we want to always set both values and treat it as a single input
+    setValue(inputName, value.input, { shouldDirty: true });
+    setValue(selectName, value.select, { shouldDirty: true });
+  };
+
+  return (
+    <>
+      <InputFieldPure className="h-7 rounded-lg border pr-0">
+        <FormField
+          control={control}
+          name={inputName}
+          render={({ field }) => (
+            <FormItem className="w-full overflow-hidden">
+              <FormControl>
+                <Input
+                  type="number"
```

Comment:
Also, when the field is focused, we should have a black border around the input.

Reply:
Validation was added.

The border issue is a bit complicated due to combination of 2 elements - will try to solve it in separate PR since its a visual issue and I dont want to block this PR to avoid further conflicts etc